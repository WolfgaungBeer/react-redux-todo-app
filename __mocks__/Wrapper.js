import React from 'react';
import { node } from 'prop-types';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // eslint-disable-line
import { Theme, defaultTheme } from '../src/components';

const mockStore = configureStore();
const store = mockStore({ scado: { theme: defaultTheme } });

const propTypes = {
    children: node,
};

const defaultProps = {
    children: undefined,
};

const Wrapper = ({ children }) => (
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

Wrapper.propTypes = propTypes;
Wrapper.defaultProps = defaultProps;

export default Wrapper;
