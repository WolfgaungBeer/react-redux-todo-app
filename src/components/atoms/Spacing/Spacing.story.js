/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import Spacing from './';


storiesOf('Spacing', module)
    .add('default', () => (
        <Spacing>Spacing</Spacing>
    ))
    .add('with props', () => (
        <Spacing
            scale={text('scale', 'l')}
            top={boolean('top', true)}
            left={boolean('left', false)}
            right={boolean('right', false)}
            bottom={boolean('bottom', false)}
        >
            Spacing
        </Spacing>
    ));
