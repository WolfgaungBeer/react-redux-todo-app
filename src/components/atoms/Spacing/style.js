export const getStyle = () => (props) => {
    const { scale, top, left, bottom, right, styles } = props;
    const size = `${props.theme.scale[scale] || props.theme.scale.m}`;
    const pos = (!top && !left && !right && !bottom) ? 'top' : undefined;
    if (pos) return `padding-${pos}: ${size}`;
    return `
        ${top && 'padding-top: '} ${size};
        ${left && 'padding-left: '} ${size};
        ${right && 'padding-right: '} ${size};
        ${bottom && 'padding-bottom: '} ${size};
        ${styles && styles(props)};
    `;
};
