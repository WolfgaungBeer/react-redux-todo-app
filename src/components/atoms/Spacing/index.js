import styled from 'styled-components';
import { string, bool, node, func } from 'prop-types';
import { getStyle } from './style';

const propTypes = {
    scale: string,
    top: bool,
    left: bool,
    bottom: bool,
    right: bool,
    className: string,
    styles: func,
    children: node,
};

const defaultProps = {
    scale: undefined,
    top: undefined,
    left: undefined,
    bottom: undefined,
    right: undefined,
    className: undefined,
    styles: undefined,
    children: undefined,
};

const Spacing = styled.div`
    ${getStyle()}
`;

Spacing.propTypes = propTypes;
Spacing.defaultProps = defaultProps;

export default Spacing;
