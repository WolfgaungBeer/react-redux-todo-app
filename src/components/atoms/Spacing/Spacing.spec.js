import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../../../__mocks__/Wrapper';
import Spacing from './';

it('renders the Spacing Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Spacing>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Spacing scale="xl">Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Spacing left>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();

    const tree4 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" top>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree4).toMatchSnapshot();

    const tree5 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" left>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree5).toMatchSnapshot();

    const tree6 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" bottom>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree6).toMatchSnapshot();

    const tree7 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" right>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree7).toMatchSnapshot();

    const tree8 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" top left right bottom>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree8).toMatchSnapshot();

    const tree9 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" styles={() => 'background-color: red;'}>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree9).toMatchSnapshot();

    const tree10 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" left right styles={() => 'background-color: red;'}>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree10).toMatchSnapshot();
});
