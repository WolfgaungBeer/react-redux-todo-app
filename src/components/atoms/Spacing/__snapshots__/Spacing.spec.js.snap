// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Spacing Component correctly 1`] = `
<div
  className="sc-bxivhb ckKdkO"
  scale={undefined}
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 2`] = `
<div
  className="sc-bxivhb ehSxoR"
  scale="xl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 3`] = `
<div
  className="sc-bxivhb lcqWmr"
  scale={undefined}
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 4`] = `
<div
  className="sc-bxivhb hJcdNe"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 5`] = `
<div
  className="sc-bxivhb LbpqH"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 6`] = `
<div
  className="sc-bxivhb hadEva"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 7`] = `
<div
  className="sc-bxivhb fzXpAn"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 8`] = `
<div
  className="sc-bxivhb PAfpz"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 9`] = `
<div
  className="sc-bxivhb gNIxJR"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 10`] = `
<div
  className="sc-bxivhb gJfcWI"
  scale="xxl"
>
  Spacing
</div>
`;
